object.go

```diff
@@ -38,7 +38,7 @@ func NewObject(ctx context.Context, fullURL string) (*Object, error) {
 
 	// store full path name and unset
 	name := strings.TrimPrefix(path.Clean(u.Path), "/")
-	u.Path = ""
+	u.Path = "/"
```

Comment:
not sure that's right...

Reply:
Well, I think... It's fine to set it to "root".

It makes it quite obvious: to resolve bucket, here's your (Resolve's) canonical bucket root directory path.

And as of current implementation it shouldn't break anything at all, as path is not really analysed in registered resolvers for each adapter (IIRC; can check tomorrow).